### ISSUE TITLE
Select expand shortest Path

### ISSUE THREAD

**Issue by boriswexler** [orientechnologies/orientdb#4663]
I posted this a while back and the answer I got was after reviewing not answering the issue, so I'd like to repost this.
Using 2.0.13

Trying <b>select expand(shortestPath) from(select shortestPath($from, $to, 'OUT') let $from = (select from User where entityId="85"), $to = (select from User where entityId="89"))</b> - doesn't give me any results.

However <b>select expand(shortestPath) from(select shortestPath(#12:8, #12:10, 'OUT')<b> works fine. 

So does <b>select shortestPath($from, $to, 'OUT') let $from = (select from User where entityId="85"), $to = (select from User where entityId="89")<b>

Can the two not be combined (i.e. can I get the expanded results in one query)? Thanks in advance for looking at this again.

Side question - can I limit the range of ShortestPath to not try beyond a certain depth in order to save processing? i.e. if I'm only interested in results up to depth 3, can I limit it to depth 3?

**Comment by luigidellaquila** [comment]
Hi @boriswexler 

It seems to be a bug
Which OrientDB version are you using? 
If you are using v 2.1 RCx you can use UNWIND operator:

```sql
 select expand(path) from(select shortestPath($from, $to, 'OUT') as path let $from = (select from User where entityId="85"), $to = (select from User where entityId="89") UNWIND path)
```

Limiting the traversal depth is not supported right now, do you mind opening another issue specifically for that? It will be easier to track.

Thanks

**Comment by boriswexler** [comment]
Thanks Luigi. I'm still on 2.09. Will upgrade to 2.1 and create another ticket

**Comment by luigidellaquila** [comment]
Hi @boriswexler 

This is already fixed in v 2.2.
For the limited depth of the path traversal, in v 2.2 we have `astar()` function that allows to do this.

I'm closing it, please reopen if you see any other problems

Thanks

Luigi
